### PR TITLE
feat(misc-apps/reloader): new chart addition

### PR DESCRIPTION
**Pull request by eyenx**
# Description

Reloader helps us perform automatically a Rolling upgrade on the relevant DeploymentConfig, Deployment, Daemonset, Statefulset and Rollout if a ConfigMap or Secret used by it is changed. 

This would help us do rolling upgrades for some apps which need to be restarted upon secret change (Example Certificate is managed by cert-manager etc.)

# Issues
n/a

# Checklist

* [x] This PR contains a description of the changes I'm making
* [x] I updated the version in Chart.yaml
* [x] I updated the changelog with an `artifacthub.io/changes` annotation in `Chart.yaml`, check the [example](docs/development.md#Changelog) in the documentation.
* [x] I updated applicable README.md files using  `pre-commit run`
* [x] I documented any high-level concepts I'm introducing in `docs/`
* [x] CI is currently green and this is ready for review
* [x] I am ready to test changes after they are applied and released

### CODE REVIEW

File: charts/misc-apps/Chart.yaml [hunk 1]
```diff
@@ -17,14 +17,8 @@ dependencies:
...
-          url: https://github.com/kedacore/keda/releases/tag/v2.10.0
-    - kind: changed
-      description: "keda: update chart from 2.9.3 to 2.10.1"
-      links:
-        - name: "feat: Ship KEDA Core v2.10.1 and align Kubernetes expectations"
-          url: https://github.com/kedacore/charts/pull/407
-        - name: "feat: prepare v2.10"
-          url: https://github.com/kedacore/charts/pull/400
+        - name: GitHub PR
+          url: https://github.com/adfinis/helm-charts/pull/xxx
```
Comment (hairmare) [inline charts/misc-apps/Chart.yaml:24]: no need for self referential linking, please add some links that help discover reloader instead

File: charts/misc-apps/examples/reloader.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# reloader watching namespaces labeled with reloader=true and exporting metrics
+
+reloader:
+  enabled: true
+  project: infra-reloader
+  values:
+    reloader:
+      namespaceSelector: "reloader=true"
```
Comment (hairmare) [inline charts/misc-apps/examples/reloader.yaml:8]: is this really a string? why doesn't it use k8s api style specs?

**Comment by hairmare** [comment]
also, can you add a description in this issue that explains what reloader is etc.

**Comment by hairmare** [comment]
oh, and this seems to be missing an update to values.yaml and README.md as well as ci/default-values.yaml.

**Comment by eyenx** [comment]
> oh, and this seems to be missing an update to values.yaml and README.md as well as ci/default-values.yaml.

YES cause I'm still working on this. Put in draft now. Sorry for the noise.

File: charts/misc-apps/examples/reloader.yaml (hunk 2 above)
Comment (eyenx) [reply charts/misc-apps/examples/reloader.yaml:11]: 6213f5c

File: charts/misc-apps/Chart.yaml (hunk 1 above)
Comment (eyenx) [reply charts/misc-apps/Chart.yaml]: 4b2ce88

File: charts/misc-apps/examples/reloader.yaml (hunk 2 above)
Comment (hairmare) [reply charts/misc-apps/examples/reloader.yaml:11]: is reloader just a kubectl wrapper? wouldn't a real k8s api client specify it as follows:
```yaml
namespaceSelector:
  reloader: "true"
```

Comment (eyenx) [reply charts/misc-apps/examples/reloader.yaml:11]: You are right. But the client expects this:

https://github.com/stakater/Reloader/blob/master/README.md?plain=1#L217

and the value is just passed over as argument in the deployment so...

**Review by gianklug — APPROVED** [review]
LGTM

**Pull request merged**